lib/msf/core/modules/metadata/search.rb

```diff
@@ -9,11 +9,61 @@ module Msf::Modules::Metadata::Search
       %w[aka author authors arch cve bid edb check date disclosure_date description fullname fullname mod_time
       name os platform path port rport rank ref ref_name reference references target targets text type]
 
+  #
+  # Module Type Shorthands
+  #
+  MODULE_TYPE_SHORTHANDS = {
+    "aux" => Msf::MODULE_AUX
+  }
+
+  #
+  # Parses command line search string into a hash. A param prefixed with '-' indicates "not", and will omit results
+  # matching that keyword. This hash can be used with the find command.
+  #
+  # Resulting Hash Example:
+  # {"platform"=>[["android"], []]} will match modules targeting the android platform
+  # {"platform"=>[[], ["android"]]} will exclude modules targeting the android platform
+  #
+  def self.parse_search_string(search_string)
+    search_string += ' '
+
+    # Split search terms by space, but allow quoted strings
+    terms = search_string.split(/\"/).collect{|term| term.strip==term ? term : term.split(' ')}.flatten
+    terms.delete('')
+
+    # All terms are either included or excluded
+    res = {}
+
+    terms.each do |term|
+      keyword, search_term = term.split(":", 2)
+      unless search_term
+        search_term = keyword
+        keyword = 'text'
+      end
+      next if search_term.length == 0
+      keyword.downcase!
+      search_term.downcase!
+
+      if keyword == "type"
+        search_term = MODULE_TYPE_SHORTHANDS[search_term] if MODULE_TYPE_SHORTHANDS.key?(search_term)
+      end
+
+      res[keyword] ||=[   [],    []   ]
+      if search_term[0,1] == "-"
+        next if search_term.length == 1
+        res[keyword][1] << search_term[1,search_term.length-1]
+      else
+        res[keyword][0] << search_term
+      end
+    end
+    res
+  end
+
   #
   # Searches the module metadata using the passed hash of search params
   #
   def find(params, fields={})
-    raise ArgumentError if params.empty? || VALID_PARAMS.none? { |k| params.key?(k) }
+    raise ArgumentError if params.any? && VALID_PARAMS.none? { |k| params.key?(k) }
```

Comment:
An empty search query will now result in all modules being returned, which aligns with the previous search functionality offered by framework's slow search / the old rpc search - https://github.com/rapid7/metasploit-framework/pull/8606

The console never went down this path, as there's a guard clause in the dispatcher:

https://github.com/rapid7/metasploit-framework/blob/2e887a8d95160a7854611c7ccd69182dfad01eb1/lib/msf/ui/console/command_dispatcher/modules.rb#L411-L418

So this feels like a safe/valid change to make 🤞